chimerapy/chimera.py

```diff
@@ -59,3 +67,149 @@ def generate_candidate_mask(m171, m193, m211):
+
+    labeled_mask = measure.label(mask)
+    regions = measure.regionprops(labeled_mask)
+
+    for region in regions:
+        if region.area < min_area_pixels:
+            labeled_mask[labeled_mask == region.label] = 0
+
+    filtered_regions = measure.regionprops(labeled_mask)
+    return labeled_mask, filtered_regions
+
+def get_coronal_holes(filtered_regions, map_obj):
+    coronal_holes = []
+
+    for region in filtered_regions:
+        coords = region.coords
+        world_coords = map_obj.pixel_to_world(coords[:, 1] * u.pix, coords[:, 0] * u.pix)
+        heliographic_coords = world_coords.transform_to("heliographic_stonyhurst")
+
+        min_lon = heliographic_coords.lon.min().deg
+        max_lon = heliographic_coords.lon.max().deg
+        min_lat = heliographic_coords.lat.min().deg
+        max_lat = heliographic_coords.lat.max().deg
+
+        extent_lon = max_lon - min_lon
+        extent_lat = max_lat - min_lat
+
+        centroid = region.centroid
+        centroid_world = map_obj.pixel_to_world(centroid[1] * u.pix, centroid[0] * u.pix)
+
+        pixel_scale_arcsec = map_obj.scale[0].value
+        pixel_scale_meters = (pixel_scale_arcsec * u.arcsec).to(u.rad).value * map_obj.rsun_meters
+        pixel_area_meters2 = pixel_scale_meters**2
+
+        cos_theta = np.abs(np.cos(heliographic_coords.lat.rad) * np.cos(heliographic_coords.lon.rad))
+        adjusted_pixel_area_meters2 = pixel_area_meters2 / cos_theta
+        adjusted_pixel_area_meters2 = np.nan_to_num(adjusted_pixel_area_meters2)
+
+        area_meters2 = np.sum(adjusted_pixel_area_meters2)
+
+        coronal_holes.append({
+            "area_pixels": region.area,
+            "area_meters2": area_meters2,
+            "centroid_world": centroid_world,
+            "centroid_heliographic": centroid_world.transform_to("heliographic_stonyhurst"),
+            "min_lon": min_lon,
+            "max_lon": max_lon,
+            "min_lat": min_lat,
+            "max_lat": max_lat,
+            "extent_lon": extent_lon,
+            "extent_lat": extent_lat,
+        })
+
+    coronal_holes = sorted(coronal_holes, key=lambda ch: ch["area_pixels"], reverse=True)
+    for idx, ch in enumerate(coronal_holes, start=1):
+        ch["id"] = idx
+
+    return coronal_holes
+
+def map_threshold(im_map):
+    """
+    Set off-disk pixels to black and clip the vmin and vmax of the map.
+
+    Parameters
+    ----------
+    im_map : `~sunpy.map.Map`
+        Unprocessed magnetogram map.
+
+    Returns
+    -------
+    im_map : `~sunpy.map.Map`
+        Processed magnetogram map.
+    """
+    im_map.data[~coordinate_is_on_solar_disk(all_coordinates_from_map(im_map))] = np.nan
+    im_map.cmap.set_bad("k")
+    im_map.plot_settings["norm"] = colors.Normalize(vmin=-200, vmax=200)
+    return im_map
+
+m171 = map_threshold(m171)
+m193 = map_threshold(m193)
+m211 = map_threshold(m211)
+
+cos_correction_171 = calculate_cosine_correction(m171)
```

Comment:
Are the values in the maps that go into the cosine correction very different, could be calculate only once?